### PR TITLE
Unit Test Mocking using Mockery

### PR DESCRIPTION
**Pull request by liamfallon**
This is a contributor guide describing how to use Mockery. It draws heavily on [PR-441](https://github.com/nephio-project/nephio/pull/441) raised by @vjayaramrh

### CODE REVIEW

**Review by vjayaramrh — APPROVED** [review]
/lgtm

File: content/en/docs/guides/contributor-guides/unit-testing-mockery.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+The targets above must be run explicitly.
+
+Run `make install-mockery` to install mockery in your container runtime (docker, podman etc) or locally if you have no container runtime running. You need only run this target once unless you need to reinstall Mockery for whatever reason.
+
+Run `make generate-mocks` to generate the mocked implementation of the go interfaces specified in '.mockery.yaml' files. You need to run this target each time an interface that you are mocking changes or whenever you change the contents of a `.mockery.yaml` file. You can run `make generate-mocks` in the repo root to generate or re-generate all interfaces or in subdirectories containing a `Makefile` to generate or regenerate only the interfaces in that subdirectory and its children.
+
+The generate-mocks target looks for `.mockery.yaml` files in the repo and it runs the mockery mock generator on each `.mockery.yaml` file it finds. This has the nice effect of allowing `.mockery.yaml` files to be in either the root of the repo or in subdirectories, so the choice of placement of `.mockery.yaml` files is left to the developer.
+
+# The .mockery.yaml file
```
Comment (Catalin-Stratulat-Ericsson) [inline content/en/docs/guides/contributor-guides/unit-testing-mockery.md:35]: just testing

File: content/en/docs/guides/contributor-guides/_index.md [hunk 2]
```diff
@@ -7,9 +7,5 @@ weight: 3
...
 # Developer guide
 
-This developer guide is for for people who want to write code for the Nephio project. This document is written as an
-extension of the [Kubernetes Developer Guides](https://github.com/kubernetes/community/tree/master/contributors/devel#the-process-of-developing-and-contributing-code-to-the-kubernetes-project)
-and therefore covers topics that are specific to Nephio development.
-
-* How to set up a [Minimal Development Environment](/content/en/docs/guides/contributor-guides/minimal-environment.md) defines common terminology
-  used in the Nephio project.
\ No newline at end of file
+This developer guide is for for people who want to write code for the Nephio project. This document is written as an extension of the [Kubernetes Developer Guides](https://github.com/kubernetes/community/tree/master/contributors/devel#the-process-of-developing-and-contributing-code-to-the-kubernetes-project)
```
Comment (electrocucaracha) [inline content/en/docs/guides/contributor-guides/_index.md:10]:
```suggestion
This developer guide is for people who want to write code for the Nephio project. This document is written as an extension of the [Kubernetes Developer Guides](https://github.com/kubernetes/community/tree/master/contributors/devel#the-process-of-developing-and-contributing-code-to-the-kubernetes-project)
```

File: content/en/docs/guides/contributor-guides/unit-testing-mockery.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+}
+type repoTest struct {
+	name    string
+	fields  fields
+	args    args
+	mocks   []mockeryutils.MockHelper
+	wantErr bool
+}
+```
+The code above allows us to specify input data and the expected outcome for tests. Each test is specified as an instance of the `repoTest` struct. Fr each test, we specify its fields and arguments, and specify the mocking for the test.
```
Comment (electrocucaracha) [inline content/en/docs/guides/contributor-guides/unit-testing-mockery.md:154]:
```suggestion
The code above allows us to specify input data and the expected outcome for tests. Each test is specified as an instance of the `repoTest` struct. For each test, we specify its fields and arguments, and specify the mocking for the test.
```

**Pull request merged**